src/ripple/shamap/SHAMap.h

```diff
@@ -182,10 +182,9 @@ class SHAMap
     
     bool getRootNode (Serializer & s, SHANodeFormat format) const;
     std::vector<uint256> getNeededHashes (int max, SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (uint256 const& hash, Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
+    SHAMapAddNode addRootNode (boost::optional<uint256> const& hash,
```

Comment:
Additionally the `hash`, when passed, is redundant information that the non-test code double-checks and reacts to with asserts (in debug) or returning error codes (in release).  The real code calls this function with the hash of a ledger.  The shamap unittest test lacks a ledger to build from.